gazelle/python/testdata/binary_without_entrypoint_per_file_generation/README.md

```diff
@@ -0,0 +1,4 @@
+# Binary without entrypoint
+
+This test case asserts that when there is no __main__.py, a py_binary is generated per file main module, and that this
+py_binary is instead of (not in addition to) any py_library target.
```

Comment:
I think your comment about the fact that py_binary can be used as a py_library should be documented in here and maybe in Gazelle plugin README. Even though this behaviour is useful, i've seen in codebases where things are maintained by hand separate py_binary and py_library targets.

Wondering that maybe it would be good to have a test of what would gazelle do in such cases?

One reason aginst relying on py_binary as tho saurce of library would be wanting to set different visibility settings.